lib/realtime/messages.ex

```diff
@@ -2,17 +2,39 @@ defmodule Realtime.Messages do
   @moduledoc """
   Handles `realtime.messages` table operations
   """
-  import Ecto.Query
-  alias Realtime.Repo
-  alias Realtime.Api.Message
 
   @doc """
   Deletes messages older than 72 hours for a given tenant connection
   """
-  @spec delete_old_messages(pid()) :: {:ok, any()} | {:error, any()}
+  @spec delete_old_messages(pid()) :: :ok
   def delete_old_messages(conn) do
-    limit = NaiveDateTime.utc_now() |> NaiveDateTime.add(-72, :hour)
-    query = from m in Message, where: m.inserted_at <= ^limit
-    Repo.del(conn, query)
+    limit =
+      NaiveDateTime.utc_now()
+      |> NaiveDateTime.add(-72, :hour)
+      |> NaiveDateTime.to_date()
+
+    %{rows: rows} =
+      Postgrex.query!(
+        conn,
+        """
+        SELECT child.relname
+        FROM pg_inherits
+        JOIN pg_class parent ON pg_inherits.inhparent = parent.oid
+        JOIN pg_class child ON pg_inherits.inhrelid = child.oid
+        JOIN pg_namespace nmsp_parent ON nmsp_parent.oid = parent.relnamespace
+        JOIN pg_namespace nmsp_child ON nmsp_child.oid = child.relnamespace
+        WHERE parent.relname = 'messages'
+        AND nmsp_child.nspname = 'realtime'
+        """,
+        []
+      )
+
+    rows
+    |> Enum.filter(fn ["messages_" <> date] ->
+      date |> String.replace("_", "-") |> Date.from_iso8601!() |> Date.compare(limit) == :lt
+    end)
+    |> Enum.each(&Postgrex.query!(conn, "DROP TABLE IF EXISTS realtime.#{&1}", []))
```

Comment:
I think this is probably fine but you should confirm that you don't need to call `truncate` first.

Reply:
added a test that does add data to all partitions. this should test if we need truncate